p2p/node_info.go

```diff
@@ -121,7 +122,7 @@ func (info DefaultNodeInfo) Validate() error {
 	// ID is already validated.
 
 	// Validate ListenAddr.
-	_, err := NewNetAddressString(IDAddressString(info.ID(), info.ListenAddr))
+	_, err := NewNetAddressString(IDAddressString(info.ID(), info.ListenAddress))
```

Comment:
should we also validate external address here?

Reply:
Probably annoying to do so since it will fail on DNS lookup. A node's external address may not be reachable from the node itself.

Reply:
We can assert that error